packages/jest-fake-timers/src/jestFakeTimers.ts

```diff
@@ -368,6 +369,12 @@ export default class FakeTimers<TimerRef> {
       // @ts-ignore TODO: figure out better typings here
       this._moduleMocker.fn().mockImplementation(impl);
 
+    const promisifiedFakeSetTimeout = fn(this._fakeSetTimeout.bind(this));
```

Comment:
Can safely just set it directly onto `this._fakeSetTimeout[util.promisify.custom]` here I think  and inline the `promisifiedFakeSetTimeout` variable down there to make it more consistent with the other `fakeTimerAPIs`

Reply:
I don't think so. The `[util.promisify.custom]` needs to be in the function passed through the `util.promisify` function. Today `global.setTimeout` is the "`fn`" function.

Reply:
I.e.
```js
    this._fakeSetTimeout[util.promisify.custom] = (
      delay?: number,
      arg?: any,
    ) => new Promise(resolve => promisifiedFakeSetTimeout(resolve, delay, arg));

    // TODO: add better typings; these are mocks, but typed as regular timers
    this._fakeTimerAPIs = {
      clearImmediate: fn(this._fakeClearImmediate.bind(this)),
      clearInterval: fn(this._fakeClearTimer.bind(this)),
      clearTimeout: fn(this._fakeClearTimer.bind(this)),
      nextTick: fn(this._fakeNextTick.bind(this)),
      setImmediate: fn(this._fakeSetImmediate.bind(this)),
      setInterval: fn(this._fakeSetInterval.bind(this)),
      setTimeout: fn(this._fakeSetTimeout.bind(this)),
    };
```

Reply:
I don't think so. The [util.promisify.custom] needs to be in the function passed through the util.promisify function. Today global.setTimeout is the "fn" function. I tried.

Reply:
Oh right, after `bind` it won't have the extra property anymore

Reply:
Nit: maybe call it `fakeSetTimeoutWithPromisifySupport` or `promisifiableFakeSetTimeout` or whatever though, since the function itself is still the normal non-promisified one :smile: